lib/rules/no-typos.js

```diff
@@ -69,7 +83,7 @@ module.exports = {
     }
 
     function checkValidPropType(node) {
-      if (node.name && !PROP_TYPES.some((propTypeName) => propTypeName === node.name)) {
+      if (node.name && !(node.name in PROP_TYPES)) {
```

Comment:
why not use Sets here rather than `in`? also, using `in` here ensures we'll get false positives for things like `toString`, or anything else on Object.prototype.

Reply:
Good catch on false positives with `in`, I definitely hadn't considered that.

Set looks like a good fit here!